### PR TITLE
feat: Integration or regression tests

### PR DESCRIPTION
**Pull request by stephantul**
This PR adds a couple of integration/regression test baselines. These are mostly meant to guard against unforeseen changes in distillation. They are not run in CI, because they download full models. I've added three models to start:

1. minilm
2. bge-base
3. embeddinggemma
4. e5-multilingual
5. gte-modernbert

These are added as defaults. You can add new models by running `make  test_integration_update`. You should only run this after changing the code.

Comparing the results without regenerating is done by running `make test_integration`. This will fail when the new results deviate significantly from the stored ones.

The integration tests surfaced a bug with BPE models and added vocabulary. An edge case, also fixed in this PR.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -58,12 +58,13 @@ dev = [
...
 ]
 
-distill = ["torch", "transformers<5.4.0", "scikit-learn", "skeletoken>=0.3.3"]
+distill = ["torch", "transformers<5.4.0", "scikit-learn", "skeletoken>=0.4.0,<0.5.0"]
 onnx = ["onnx", "torch"]
 # train also installs inference
 train = ["torch", "lightning", "scikit-learn", "skops"]
 inference = ["scikit-learn", "skops"]
 quantization = ["scikit-learn"]
+integration = ["mteb"]
```
Comment (Pringled) [inline pyproject.toml:67]: uv lock needs to be regenerated I think

**Review by Pringled — APPROVED** [review]
LGTM, nice!
